engine/src/main/scala/cromwell/engine/workflow/lifecycle/WorkflowExecutionActor.scala

```diff
@@ -45,32 +97,56 @@ object WorkflowExecutionActor {
   case object WorkflowExecutionAbortedResponse extends WorkflowExecutionActorResponse
   final case class WorkflowExecutionFailedResponse(reasons: Seq[Throwable]) extends WorkflowExecutionActorResponse
 
+  /**
+    * Captures the `ExecutionStoreKey`/`ExecutionStatus` combination for a call that has been created or started
+    * as part of `startRunnableCalls`.  These are tracked to coordinate processing of messages based on completion
+    * of persistence.
+    */
+  case class StartEntry(execution: JobKey, status: ExecutionStatus)
+
+  case class ExecutionStartResult(startEntries: Traversable[StartEntry],
+                                  newEntries: Traversable[JobKey] = Seq.empty) {
+    /** Merge two `ExecutionStartResult`s. */
+    def plus(other: ExecutionStartResult): ExecutionStartResult = {
+      ExecutionStartResult(startEntries = startEntries ++ other.startEntries, newEntries = newEntries ++ other.newEntries)
+    }
+  }
+
   def props(workflowId: WorkflowId, workflowDescriptor: EngineWorkflowDescriptor): Props = Props(WorkflowExecutionActor(workflowId, workflowDescriptor))
 }
 
 final case class WorkflowExecutionActor(workflowId: WorkflowId, workflowDescriptor: EngineWorkflowDescriptor) extends LoggingFSM[WorkflowExecutionActorState, WorkflowExecutionActorData] {
 
+  import WorkflowExecutionActor._
+
   val tag = self.path.name
-  startWith(WorkflowExecutionPendingState, WorkflowExecutionActorData())
 
-  def startJob(call: Call) = {
+  private val calls = workflowDescriptor.backendDescriptor.workflowNamespace.workflow.calls
+
+  // Initialize the StateData with ExecutionStore (all calls as NotStarted) and SymbolStore
+  startWith(WorkflowExecutionPendingState,
+    WorkflowExecutionActorData(executionStore = (calls map (BackendJobDescriptorKey(_, None, 1) -> NotStarted)) toMap,
+      symbolCache = buildSymbolStoreEntries.groupBy(entry => SymbolCacheKey(entry.scope, entry.isInput))))
+
+  def startJob(backendJobDescriptorKey: BackendJobDescriptorKey, stateData: WorkflowExecutionActorData): Try[ExecutionStartResult] = {
     // TODO: Support indexes and retries:
-    val jobKey = BackendJobDescriptorKey(call, None, 1)
-    val jobDescriptor: BackendJobDescriptor = BackendJobDescriptor(workflowDescriptor.backendDescriptor, jobKey, Map.empty)
-    val executionActor = backendForExecution(jobDescriptor, workflowDescriptor.backendAssignments(call))
-    executionActor ! ExecuteJobCommand
+    val call = backendJobDescriptorKey.call
+    fetchLocallyQualifiedInputs(backendJobDescriptorKey, stateData.symbolCache) match {
+      case Success(symbolStoreForCall) =>
+        val jobDescriptor: BackendJobDescriptor = BackendJobDescriptor(workflowDescriptor.backendDescriptor, backendJobDescriptorKey, symbolStoreForCall)
+        val executionActor = backendForExecution(jobDescriptor, workflowDescriptor.backendAssignments(call))
```

Comment:
You might want to coordinate with @scottfrazer on how to wire up the actor factory properly (I believe his PR is about to merge)